packages/playground/src/utils/ssh.ts

```diff
@@ -218,6 +251,31 @@ class SSHKeysManagement {
   availablePublicKey(publicKey: string): boolean {
     return !this.list().some(key => key.publicKey === publicKey);
   }
+  needsDefaultNameAssignment(keys?: SSHKeyData[]): boolean {
+    if (!keys) {
+      keys = this.oldKeys as SSHKeyData[];
+    }
+    return keys.some(key => isAlphanumericWithSpace("Invalid name")(key.name) !== true);
+  }
+
+  /**
+   * Checks all user keys for empty names and assigns a unique default name (default, default1, ...).
+   * Updates the keys in-place and returns the updated array.
+   * @param keys The SSH keys to check and update.
+   * @returns The updated array of SSH keys.
+   */
+  assignDefaultNames(keys?: SSHKeyData[]): SSHKeyData[] {
+    if (!keys) {
+      keys = this.oldKeys as SSHKeyData[];
+    }
+    const existingNames = new Set(keys.map(k => k.name).filter(Boolean));
+    for (const key of keys) {
+      if (isAlphanumericWithSpace("Invalid name")(key.name) !== true) {
+        key.name = this.getUniqueName(existingNames, false);
+      }
+    }
+    return keys;
+  }
```

Comment:
```suggestion
 assignDefaultNames(keys?: SSHKeyData[]): SSHKeyData[] {
  if (!keys) {
    keys = this.oldKeys as SSHKeyData[];
  }

  // Collect existing names, filtering out empty or falsy names
  const existingNames = new Set(
    keys
      .map(k => k.name)
      .filter(name => name && isAlphanumericWithSpace("Invalid name")(name) === true)
  );

  for (const key of keys) {
    // Check if name is invalid or empty
    if (!key.name || isAlphanumericWithSpace("Invalid name")(key.name) !== true) {
      // Assign a unique default name
      key.name = this.getUniqueName(existingNames, false);
    } else {
      // Add valid existing name to the set to prevent duplicates
      existingNames.add(key.name);
    }
  }
  return keys;
}
```